File: internal/sast/fixtures/javascript/nosql_positive.js
// internal/sast/fixtures/javascript/nosql_positive.js
const express = require("express");
const User = require("./models/user");
const app = express();

app.post("/login", async (req, res) => {
  const user = await User.findOne({ username: req.body.username, password: req.body.password });
  res.json({ success: !!user });
});

File: internal/sast/fixtures/python/nosql_positive.py
# internal/sast/fixtures/python/nosql_positive.py
from flask import Flask, request, jsonify
from pymongo import MongoClient

app = Flask(__name__)
db = MongoClient()["app"]

@app.route("/login", methods=["POST"])
def login():
    user = db.users.find_one({"username": request.json["username"], "password": request.json["password"]})
    return jsonify(user is not None)
